### PR TITLE
Draft: #477 Add a VID field for non-VID scoped PAAs (Root certificates) 

### PR DESCRIPTION
**Pull request by Toktar**
https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/477

### CODE REVIEW

File: types/pki/errors.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ var (
 	ErrInvalidVidFormat                                  = sdkerrors.Register(ModuleName, 432, "invalid vid format")
 	ErrInvalidPidFormat                                  = sdkerrors.Register(ModuleName, 433, "invalid pid format")
 	ErrInvalidDataURLFormat                              = sdkerrors.Register(ModuleName, 434, "invalid data url format")
+	ErrCertificateVidNotEqualMsgVid                      = sdkerrors.Register(ModuleName, 435, "certificate vid does not equal message vid")
```
[the remark this answers is not in the document]
Comment (Toktar) [reply types/pki/errors.go:45]: Do we also want to change other error messages?
https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/486/files#diff-201e80de8b4c427b08bcde506b0bb2bda9997e18be9975994fa60513d4990c7bR25-R28

File: integration_tests/cli/pki-demo.sh [hunk 2]
```diff
@@ -1647,12 +1648,12 @@ test_divider
...
 echo "$user_account (Not Trustee) propose Root certificate"
 test_root_path="integration_tests/constants/test_root_cert"
-result=$(echo "$passphrase" | dcld tx pki propose-add-x509-root-cert --certificate="$test_root_path" --from $user_account --yes)
+result=$(echo "$passphrase" | dcld tx pki propose-add-x509-root-cert --certificate="$test_root_path" --vid $vid --from $user_account --yes)
 response_does_not_contain "$result" "\"code\": 0"
 
 echo "$trustee_account (Trustee) propose Root certificate"
 test_root_path="integration_tests/constants/test_root_cert"
-result=$(echo "$passphrase" | dcld tx pki propose-add-x509-root-cert --certificate="$test_root_path" --from $trustee_account --yes)
+result=$(echo "$passphrase" | dcld tx pki propose-add-x509-root-cert --certificate="$test_root_path" --vid $vid --from $trustee_account --yes)
```
[the remark this answers is not in the document]
Comment (Toktar) [reply integration_tests/cli/pki-demo.sh]: done

File: x/pki/types/message_propose_add_x_509_root_cert.go [hunk 3]
```diff
@@ -56,5 +59,14 @@ func (msg *MsgProposeAddX509RootCert) ValidateBasic() error {
 		return err
 	}
 
+	cert, err := x509.DecodeX509Certificate(msg.Cert)
+	if err != nil {
+		return pkitypes.NewErrInvalidCertificate(err)
+	}
+	subjectVid, err := x509.GetVidFromSubject(cert.SubjectAsText)
+	if err == nil && subjectVid != 0 && subjectVid != msg.Vid {
+		return pkitypes.NewErrCertificateVidNotEqualMsgVid(fmt.Sprintf("Certificate VID=%d does not equal msg VID=%d", subjectVid, msg.Vid))
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/pki/types/message_propose_add_x_509_root_cert.go]: done

File: x/pki/types/message_propose_add_x_509_root_cert.go [hunk 4]
```diff
@@ -56,5 +59,14 @@ func (msg *MsgProposeAddX509RootCert) ValidateBasic() error {
 		return err
 	}
 
+	cert, err := x509.DecodeX509Certificate(msg.Cert)
+	if err != nil {
+		return pkitypes.NewErrInvalidCertificate(err)
+	}
+	subjectVid, err := x509.GetVidFromSubject(cert.SubjectAsText)
+	if err == nil && subjectVid != 0 && subjectVid != msg.Vid {
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/pki/types/message_propose_add_x_509_root_cert.go]: done

File: x/pki/types/message_propose_add_x_509_root_cert_test.go [hunk 5]
```diff
@@ -47,9 +48,21 @@ func TestMsgProposeAddX509RootCert_ValidateBasic(t *testing.T) {
...
+			name: "VID is required",
+			msg: MsgProposeAddX509RootCert{
+				Signer: sample.AccAddress(),
+				Cert:   testconstants.RootCertPem,
+				Info:   testconstants.Info,
+				Time:   12345,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
 	}
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/pki/types/message_propose_add_x_509_root_cert_test.go:78]: done

File: types/pki/errors.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply types/pki/errors.go]: Yes, would be great to change it in the next PRs

**Pull request closed without merging**